alphafold3_pytorch/inputs.py

```diff
@@ -3412,15 +3414,34 @@ def __getitem__(self, idx: int | str) -> PDBInput:
         if self.training:
             cropping_config = self.cropping_config
 
-        pdb_input = PDBInput(
+        i = PDBInput(
             mmcif_filepath=str(mmcif_filepath),
             chains=(chain_id_1, chain_id_2),
             cropping_config=cropping_config,
             training=self.training,
             **self.pdb_input_kwargs,
         )
 
-        return pdb_input
+        if self.return_atom_inputs:
+            i = maybe_transform_to_atom_input(i)
+        
+        return i
+
+    def __getitem__(self, idx: int | str, max_attempts: int = 10) -> Union[PDBInput, AtomInput]:
+        """Return either a PDBInput or an AtomInput object for the specified index."""
+        i = self.get_item(idx)
+
+        # if the input could not be successfully constructed, try a new example up to `max_attempts` times
+        if exists(i):
+            return i
+        else:
+            attempts = 0
```

Comment:
this is actually a very common pattern

i think it is worth adding a small dep just to avoid rewriting this

perhaps [this library](https://github.com/groodt/retrying)? this one can retry on a result of `None`

Reply:
just tested it, seems to work fine

```python
# at top

from retrying import retry

def not_exists(v):
    return v is None

# within the function

retry_decorator = retry(retry_on_result=not_exists, stop_max_attempt_number = 3)
i = retry_decorator(self.get_item)(idx)
```

Reply:
I'll add this now

Reply:
oh and `retrying` will have to be added [here](https://github.com/lucidrains/alphafold3-pytorch/blob/main/pyproject.toml#L48)

Reply:
This get_item function looks so much cleaner now

Reply:
decorators are def one of the most elegant features of python